src/ServicePulse.Host/vue/src/views/MessageView.vue

```diff
@@ -32,6 +32,49 @@ function loadFailedMessage() {
     });
 }
 
+function downloadHeadersAndBody() {
+    return useFetchFromServiceControl("messages/search/" + id)
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+        if (data[0] === undefined) {
+            failedMessage.value.headersNotFound = true ;
+            failedMessage.value.messageBodyNotFound = true;
+            return;
+        }
+      var message = data[0];
+      var headers = message.headers;
+      failedMessage.value.headers = headers;
+      downloadBody();
```

Comment:
```suggestion
      if (data[0] === undefined) {
          failedMessage.value.headersNotFound = true ;
          failedMessage.value.messageBodyNotFound = true;
          return;
      }
      var message = data[0];
      var headers = message.headers;
      failedMessage.value.headers = headers;
      return downloadBody();
```

Weird formatting.

Also, don't forget to return the promise of `downloadBody`